build-logic/src/main/kotlin/com/apollographql/apollo3/buildlogic/plugin/LibraryConventionPlugin.kt

```diff
@@ -0,0 +1,67 @@
+package com.apollographql.apollo3.buildlogic.plugin
+
+import configureJavaAndKotlinCompilers
+import configureMppDefaults
+import configurePublishing
+import configureRepositories
+import configureTesting
+import org.gradle.api.Action
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Nested
+import org.gradle.api.tasks.bundling.Jar
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+class LibraryConventionPlugin : Plugin<Project> {
+  override fun apply(project: Project) {
+    with(project) {
+      group = property("GROUP")!!
+      version = property("VERSION_NAME")!!
+
+      configureRepositories()
+
+      extensions.create("apolloLibrary", Extension::class.java)
+
+      configureJavaAndKotlinCompilers(treatWarningsAsErrors = true)
+
+      configureTesting()
+
+      configurePublishing()
+    }
+  }
+
+  abstract class Extension(private val project: Project) {
+    interface MppConfiguration {
+      val withJs: Property<Boolean>
+      val withLinux: Property<Boolean>
+    }
+
+    @get:Nested
+    abstract val mppConfiguration: MppConfiguration
+
+    fun mpp(action: Action<MppConfiguration>) {
+      action.execute(mppConfiguration)
+      project.configureMppDefaults(
+          withJs = mppConfiguration.withJs.getOrElse(true),
+          withLinux = mppConfiguration.withLinux.getOrElse(true)
+      )
+    }
+
+    fun javaModuleName(javaModuleName: String) {
```

Comment:
Had to make this one a function instead of a Property in the end, otherwise it was (on jvm modules only!) called too early when the value of the property wasn't available uet.